x-pack/legacy/plugins/transform/public/app/sections/clone_transform/clone_transform_section.tsx

```diff
@@ -0,0 +1,194 @@
+import { breadcrumbService, docTitleService, BREADCRUMB_SECTION } from '../../services/navigation';
+import { documentationLinksService } from '../../services/documentation';
+import { PrivilegesWrapper } from '../../lib/authorization';
+import {
+  getIndexPatternIdByTitle,
+  loadIndexPatterns,
+  KibanaProvider,
+  RenderOnlyWithInitializedKibanaContext,
+} from '../../lib/kibana';
+
+import { Wizard } from '../create_transform/components/wizard';
+
+const indexPatterns = npStart.plugins.data.indexPatterns;
+const savedObjectsClient = npStart.core.savedObjects.client;
+
+interface GetTransformsResponseOk {
+  count: number;
+  transforms: TransformPivotConfig[];
+}
+
+interface GetTransformsResponseError {
+  error: {
+    msg: string;
+    path: string;
+    query: any;
+    statusCode: number;
+    response: string;
+  };
+}
+
+function isGetTransformsResponseError(arg: any): arg is GetTransformsResponseError {
+  return arg.error !== undefined;
+}
+
+type GetTransformsResponse = GetTransformsResponseOk | GetTransformsResponseError;
+
+type Props = RouteComponentProps<{ transformId: string }>;
+export const CloneTransformSection: FC<Props> = ({ match }) => {
+  // Set breadcrumb and page title
+  useEffect(() => {
+    breadcrumbService.setBreadcrumbs(BREADCRUMB_SECTION.CLONE_TRANSFORM);
+    docTitleService.setTitle('createTransform');
+  }, []);
+
+  const api = useApi();
+
+  const transformId = match.params.transformId;
+
+  const [transformConfig, setTransformConfig] = useState<TransformPivotConfig>();
+  const [errorMessage, setErrorMessage] = useState();
+  const [isInitialized, setIsInitialized] = useState(false);
+  const [savedObjectId, setSavedObjectId] = useState<string | undefined>(undefined);
+
+  const fetchTransformConfig = async () => {
+    try {
+      const transformConfigs: GetTransformsResponse = await api.getTransforms(transformId);
+      if (isGetTransformsResponseError(transformConfigs)) {
+        setTransformConfig(undefined);
+        setErrorMessage(transformConfigs.error.msg);
+        setIsInitialized(true);
+        return;
+      }
+
+      await loadIndexPatterns(savedObjectsClient, indexPatterns);
+      const indexPatternTitle = Array.isArray(transformConfigs.transforms[0].source.index)
+        ? transformConfigs.transforms[0].source.index.join(',')
+        : transformConfigs.transforms[0].source.index;
+      const indexPatternId = getIndexPatternIdByTitle(indexPatternTitle);
+
+      if (indexPatternId === undefined) {
+        throw new Error(
+          i18n.translate('xpack.transform.clone.errorPromptText', {
+            defaultMessage: 'Could not fetch the Kibana index pattern ID.',
+          })
+        );
+      }
+
+      setSavedObjectId(indexPatternId);
+
+      setTransformConfig(transformConfigs.transforms[0]);
+      setErrorMessage(undefined);
+      setIsInitialized(true);
+    } catch (e) {
+      setTransformConfig(undefined);
+      if (e.message !== undefined) {
```

Comment:
I see you have `isGetTransformsResponseError `, maybe it should be here instead?

Reply:
This `catch` is about catching any kind of error that could happen in the `try` block, more code related than response related. `isGetTransformsResponseError` is about passing on `transformConfigs.error.msg` where this is `e.message`.